autosklearn/data/validation.py

```diff
@@ -26,603 +21,115 @@ class InputValidator:
-        y = self.validate_target(y, is_classification)
-
-        if X.shape[0] != y.shape[0]:
-            raise ValueError(
-                "The number of samples from the features X={} should match "
-                "the number of samples from the target y={}".format(
-                    X.shape[0],
-                    y.shape[0]
-                )
-            )
-        return X, y
-
-    def validate_features(
-        self,
-        X: Union[pd.DataFrame, np.ndarray],
-    ) -> np.ndarray:
-        """
-        Wrapper around sklearn check_array. Translates a pandas
-        Dataframe to a valid input for sklearn.
-        """
+        self.feature_validator = FeatureValidator()
+        self.target_validator = TargetValidator()
+        self._is_fitted = False
 
-        # Make sure that once fitted, we don't allow new dtypes
-        if self.features_type is None:
-            self.features_type = type(X)
-        if self.features_type != type(X):
-            raise ValueError("Auto-sklearn previously received features of type {} "
-                             "yet the current features have type {}. Changing the dtype "
-                             "of inputs to an estimator is not supported.".format(
-                                    self.features_type,
-                                    type(X)
-                                )
-                             )
-
-        # Do not support category/string numpy data. Only numbers
-        if hasattr(X, "dtype") and not np.issubdtype(X.dtype.type, np.number):
-            raise ValueError(
-                "When providing a numpy array to Auto-sklearn, the only valid "
-                "dtypes are numerical ones. The provided data type {} is not supported."
-                "".format(
-                    X.dtype.type,
-                )
-            )
-
-        # Pre-process dataframe to make them numerical
-        # Also, encode numpy categorical objects
-        if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
-            # Pandas validation provide extra user information
-            X = self._check_and_encode_features(X)
-
-        if scipy.sparse.issparse(X):
-            X.sort_indices()
-
-        # sklearn check array will make sure we have the
-        # correct numerical features for the array
-        # Also, a numpy array will be created
-        X = sklearn.utils.check_array(
-            X,
-            force_all_finite=False,
-            accept_sparse='csr'
-        )
-        return X
-
-    def validate_target(
+    def fit(
         self,
-        y: Union[pd.DataFrame, np.ndarray],
+        X_train: typing.Union[np.ndarray, pd.DataFrame],
+        y_train: typing.Union[np.ndarray, pd.DataFrame],
+        X_test: typing.Optional[typing.Union[np.ndarray, pd.DataFrame]] = None,
+        y_test: typing.Optional[typing.Union[np.ndarray, pd.DataFrame]] = None,
+        feat_type: typing.Optional[typing.List[str]] = None,
```

Comment:
Would it be possible to move `feat_type` and `is_classification` to the constructor?

Reply:
Done!